### ISSUE TITLE
panic: Crashes when Visitor visiting Object deconstructing pattern

### ISSUE THREAD

**Issue by Jack-Works** [swc-project/swc#1206]
```js
const { transformSync } = require("@swc/core");
const Visitor = require("@swc/core/Visitor").default;

transformSync(
    `
const {x} = y
`,
    { plugin: (p) => new Visitor().visitProgram(p) }
);
```

Crashes:

```
        return nodes.map(this.visitObjectPatternProperty.bind(this));
                     ^

TypeError: Cannot read property 'map' of undefined
    at Visitor.visitObjectPatternProperties (@swc/core@1.2.37/node_modules/@swc/core/Visitor.js:1248:22)
    at Visitor.visitObjectPattern (@swc/core@1.2.37/node_modules/@swc/core/Visitor.js:1243:24)
    at Visitor.visitPattern (@swc/core@1.2.37/node_modules/@swc/core/Visitor.js:1222:29)
    at Visitor.visitVariableDeclarator (@swc/core@1.2.37/node_modules/@swc/core/Visitor.js:404:21)
    at Array.map (<anonymous>)
    at Visitor.visitVariableDeclarators (@swc/core@1.2.37/node_modules/@swc/core/Visitor.js:401:22)
    at Visitor.visitVariableDeclaration (@swc/core@1.2.37/node_modules/@swc/core/Visitor.js:397:31)
    at Visitor.visitDeclaration (@swc/core@1.2.37/node_modules/@swc/core/Visitor.js:393:29)
    at Visitor.visitStatement (@swc/core@1.2.37/node_modules/@swc/core/Visitor.js:218:29)
    at Visitor.visitModuleItem (@swc/core@1.2.37/node_modules/@swc/core/Visitor.js:36:29)
```

swc version: 1.2.37

**Comment by grosto** [comment]
This seems to be fixed on 1.2.49. At least with default config

**Comment by kdy1** [comment]
Thanks!
